landlab/components/carbonate/carbonate_producer.py

```diff
@@ -0,0 +1,219 @@
+#! /usr/bin/env python
+from contextlib import contextmanager
+
+import numpy as np
+from landlab import Component
+
+
+@contextmanager
+def set_numpy_err(*args, **kwds):
+    settings = np.seterr(*args, **kwds)
+    try:
+        yield settings
+    finally:
+        np.seterr(**settings)
+
+
+def smooth_heaviside(x, width=0.5, out=None):
+    if width > 0.0:
+        with set_numpy_err(over="ignore"):
+            return np.divide(1.0, (1.0 + np.exp(-2.0 / width * np.asarray(x))), out=out)
+    else:
+        return np.heaviside(x, 0.5)
+
+
+class CarbonateProducer(Component):
+    """When I properly understand how to write the code, I'll write this intro ;-)"""
+
+    _name = "The Producer of Carbonate"
+
+    _unit_agnostic = True
+
+    _info = {
+        "sea_level__elevation": {
+            "dtype": "float",
+            "intent": "in",
+            "optional": False,
+            "units": "m",
+            "mapping": "grid",
+            "doc": "Position of sea level",
+        },
+        "topographic__elevation": {
+            "dtype": "float",
+            "intent": "inout",
+            "optional": False,
+            "units": "m",
+            "mapping": "node",
+            "doc": "Land surface topographic elevation",  # and seafloor
+        },
+        "carbonate_production_rate": {
+            "dtype": "float",
+            "intent": "out",
+            "optional": False,
+            "units": "m / y",
+            "mapping": "node",
+            "doc": "Carbonate production rate in latest time step",
+        },
+    }
+
+    def __init__(
+        self,
+        grid,
+        extinction_coefficient=1.0,
+        max_carbonate_production_rate=1000.0,
+        saturating_light=1.0,
+        surface_light=1.0,
+        tidal_range=0.0,
+    ):
+        """
+        Parameters
+        ----------
+        grid: ModelGrid (RasterModelGrid, HexModelGrid, etc.)
+            A landlab grid.
+        """
+        self.extinction_coefficient = extinction_coefficient

```

Comment:
Shouldn't these be private variables, since they have public getters and setters defined below?